pkg/handler/v2/external_cluster/external_cluster.go

```diff
@@ -1038,7 +1039,7 @@ func convertClusterToAPIWithStatus(ctx context.Context, clusterProvider provider
 	apiCluster.Status = status
 	cloud := internalCluster.Spec.CloudSpec
 	kubeOneCondtion := internalCluster.Status.Condition
-	if cloud == nil {
+	if cloud == nil || cloud.ProviderName == "" {
```

Comment:
plz change this to 
`if cloud != nil && cloud.ProviderName == ""`

Reply:
Why ? `cloud` will never actually equals to nil meaning that the second condition will always hit

Reply:
i know, but in cases user creates a external cluster crd manually in the cluster

Reply:
The one that I added during the huddle

Reply:
Does CloudSpec need to be a pointer? If it's never supposed to be nil.

Reply:
Until now it is not clear how to treat this, thus I would like to keep a pointer. Once there are more of a concrete idea of how to handle external cluster we can revisit this again.